### DISCUSSION TITLE
DataTable - Add Row with Totals

### DISCUSSION THREAD

**Discussion by egorshubin** [primefaces/community#2576] [PrimeVue]
Is it possible to add row with totals? E.g. containing sum of Revenue, avg percent. Right now if I have e.g. 10 rows, I show only 9 rows of ten, and the 10nth row goes for totals, which is not pretty and is a crunch

**Comment by auzair92** [comment]
I believe there is a "footer" template for Table component which you can use for this purpose. From the Table documentation:

```
    <ng-template pTemplate="footer">
        <tr>
            <td colspan="3" class="text-right">Totals</td>
            <td>{{lastYearTotal | currency: 'USD'}}</td>
            <td>{{thisYearTotal | currency: 'USD'}}</td>
        </tr>
    </ng-template>
```

![Screenshot_26-8-2024_105637_primeng org](https://github.com/user-attachments/assets/8713abd1-6089-4585-8d46-e66c4bcc9333)

Is this what you're looking for?

**Reply by egorshubin** [reply]
I'd rather have an already created solution, because writing such a template is suitable if you have constant columns in your table. I have over 70 columns, that can be added or removed, their place can change, and I want totals that I count by a separate axios request to be displayed under corresponding columns.
![Screenshot_89](https://github.com/user-attachments/assets/74be70a6-24b9-4b3f-b1f7-0104e355fdbe)
What I did is that I lazy-load e.g. 9 rows instead of 10, then I lazy-load one more row with totals and add it to rows array in store. Then it's reactively displayed. Then I apply styles to the row. This variant suits me. But it would be better to have a setting for such a totals row that is consistent with flexible columns order
